paddle/fluid/framework/ir/graph.cc

```diff
@@ -208,7 +216,8 @@ Graph::Graph(const ProgramDesc &program) : program_(program) {
       // Add write after write dependence
       ir::Node *upstream_op =
           (*it_old)->inputs.empty() ? nullptr : (*it_old)->inputs[0];
-      if (upstream_op) {
+      // TODO(zcd): Add a test.
+      if (upstream_op && upstream_op != write_op) {
```

Comment:
Is it necessary that adding `upstream_op != write_op`?
From my understanding, for SSA-Granph, a version of a variable is only generated by one `write_op`, if in some situation, upstream_op is equal to write_op, it is to say `write_op` generates a variable with two different version, I think it violates the conception of SSA-Granph.

Reply:
I hit an error if I don't do this

In while_op_grad, there are multiple output with the same name @EMPTY@. Then you will connect the same op to itself, resulting in loops

Reply:
Thanks, I see.